src/Symfony/Component/Console/Application.php

```diff
@@ -941,6 +959,17 @@ protected function doRunCommand(Command $command, InputInterface $input, OutputI
             // ignore invalid options/arguments for now, to allow the event listeners to customize the InputDefinition
         }
 
+        if ($this->signalRegistry) {
+            foreach ($this->handlingSignals as $handlingSignal) {
+                $event = new ConsoleSignalEvent($command, $input, $output, $handlingSignal);
+                $onSignalHandler = function () use ($event) {
+                    $this->dispatcher->dispatch($event, ConsoleEvents::SIGNAL);
+                };
+
+                $this->signalRegistry->register($handlingSignal, $onSignalHandler);
+            }
+        }
```

Comment:
signal might happen before that `doRunCommand()` gets reached (when you don't have a running command yet). Should handlers registration be done earlier? e.g. in `doRun()`?